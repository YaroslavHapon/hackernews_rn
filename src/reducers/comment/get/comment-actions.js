export const FETCH_COMMENT_STARTED = 'FETCH_COMMENT';
export const FETCH_COMMENT_SUCCESS = 'FETCH_COMMENT_SUCCESS';
export const FETCH_COMMENT_FAILURE = 'FETCH_COMMENT_FAILURE';

export function fetchComment (id) {
  return {
    type: FETCH_COMMENT_STARTED,
    payload: {
      id,
      data: {}
    }
  }
}