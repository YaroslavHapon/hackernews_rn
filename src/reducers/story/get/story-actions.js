export const FETCH_STORY_STARTED = 'FETCH_STORY';
export const FETCH_STORY_SUCCESS = 'FETCH_STORY_SUCCESS';
export const FETCH_STORY_FAILURE = 'FETCH_STORY_FAILURE';

export function fetchStory (id) {
  return {
    type: FETCH_STORY_STARTED,
    payload: {
      id
    }
  }
}