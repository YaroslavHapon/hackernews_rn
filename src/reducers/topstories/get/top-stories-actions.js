export const FETCH_TOP_STORIES_STARTED = 'FETCH_TOP_STORIES';
export const FETCH_TOP_STORIES_SUCCESS = 'FETCH_TOP_STORIES_SUCCESS';
export const FETCH_TOP_STORIES_FAILURE = 'FETCH_TOP_STORIES_FAILURE';

export function fetchTopStories () {
  return {
    type: FETCH_TOP_STORIES_STARTED,
    payload: []
  }
}