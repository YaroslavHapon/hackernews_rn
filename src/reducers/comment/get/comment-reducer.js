import * as actions from './comment-actions'

const initialState = {
  loading: false,
  loaded: false,
  data: null,
  error: null
};

export default function commentsGetReducer(state = {}, action) {
  switch (action.type) {
    case actions.FETCH_COMMENT_STARTED:
      return {
        ...state,
        [action.payload.id]: {
          ...initialState,
          loading: true
        }
      };
    case actions.FETCH_COMMENT_SUCCESS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          loading: false,
          loaded: true,
          data: action.payload.data
        }
      };
    case actions.FETCH_COMMENT_FAILURE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          loading: false,
          error: action.payload.error
        }
      }
  }
  return state;
}