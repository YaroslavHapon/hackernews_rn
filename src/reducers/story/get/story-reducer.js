import * as actions from './story-actions'

export const initialState = {
  loading: false,
  loaded: false,
  data: null,
  error: null
};

export default function storyGetReducer(state = {}, action) {
  switch (action.type ) {
    case actions.FETCH_STORY_STARTED:
      return {
        ...state,
        [action.payload.id]: {
          ...initialState,
          loading: true
        }
      }
    case actions.FETCH_STORY_SUCCESS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          loading: false,
          loaded: true,
          data: action.payload.data
        }
      }
    case actions.FETCH_STORY_FAILURE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          loading: false,
          error: action.payload.error
        }
      }
  }
  return state;
}