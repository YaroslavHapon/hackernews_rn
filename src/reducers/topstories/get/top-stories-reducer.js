import * as actions from './top-stories-actions'

const initialState = {
  loading: false,
  loaded: false,
  data: null,
  error: null
};

export default function topStoriesGetReducer (state= initialState, action) {
  switch (action.type) {
    case actions.FETCH_TOP_STORIES_STARTED:
      return { ...state, loading: true };
    case actions.FETCH_TOP_STORIES_SUCCESS:
      return { ...state, loading: false, loaded: true, data: action.payload };
    case actions.FETCH_TOP_STORIES_FAILURE:
      return {...state, loading: false, loaded: false, error: action.payload };
  }
  return state
}