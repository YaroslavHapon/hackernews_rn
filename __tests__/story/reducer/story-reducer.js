import storyGetReducer from  './../../../src/reducers/story/get/story-reducer'
import * as actions from './../../../src/reducers/story/get/story-actions'

describe('Testing Story reducer', () => {
  it('should handles request start action', () => {
    const startRequestAction = {
      type: actions.FETCH_STORY_STARTED,
      payload: { id: 1234 }
    };

    expect(storyGetReducer({}, startRequestAction)).toMatchSnapshot();
  });

  it('should handles request start action', () => {
    const startRequestAction = {
      type: actions.FETCH_STORY_SUCCESS,
      payload: { id: 1234, data: { title: "New story" } }
    };

    expect(storyGetReducer({}, startRequestAction)).toMatchSnapshot();
  });

  it('should handles request start action', () => {
    const startRequestAction = {
      type: actions.FETCH_STORY_FAILURE,
      payload: { id: 1234, error: { message: 'Story request fails!'} }
    };

    expect(storyGetReducer({}, startRequestAction)).toMatchSnapshot();
  });
});