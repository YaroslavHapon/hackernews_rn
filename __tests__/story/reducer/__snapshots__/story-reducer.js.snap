// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Story reducer should handles request start action 1`] = `
Object {
  "1234": Object {
    "data": null,
    "error": null,
    "loaded": false,
    "loading": true,
  },
}
`;

exports[`Testing Story reducer should handles request start action 2`] = `
Object {
  "1234": Object {
    "data": Object {
      "title": "New story",
    },
    "loaded": true,
    "loading": false,
  },
}
`;

exports[`Testing Story reducer should handles request start action 3`] = `
Object {
  "1234": Object {
    "error": Object {
      "message": "Story request fails!",
    },
    "loading": false,
  },
}
`;
